modules/tensor_mechanics/src/materials/RadialReturnStressUpdate.C

```diff
@@ -106,6 +110,39 @@ RadialReturnStressUpdate::updateState(RankTwoTensor & strain_increment,
   stress_new = elasticity_tensor * (strain_increment + elastic_strain_old);
 
   computeStressFinalize(inelastic_strain_increment);
+
+  if (compute_full_tangent_operator &&
+      getTangentCalculationMethod() == TangentCalculationMethod::PARTIAL)
+  {
+    if (MooseUtils::absoluteFuzzyEqual(scalar_effective_inelastic_strain, 0.0))
+      tangent_operator.zero();
+    else
+    {
+      // mu = _three_shear_modulus / 3.0;
+      // norm_dev_stress = ||s_n+1||
+      // effective_trial_stress = von mises trial stress = std::sqrt(3.0 / 2.0) * ||s_n+1^trial||
+      // scalar_effective_inelastic_strain = Delta epsilon^cr_n+1
+      // deriv = derivative of scalar_effective_inelastic_strain w.r.t. von mises stress
+      // deriv = std::sqrt(3.0 / 2.0) partial Delta epsilon^cr_n+1n over partial ||s_n+1^trial||
+
+      mooseAssert(_three_shear_modulus != 0.0, "Shear modulus is zero");
+
+      const RankTwoTensor deviatoric_stress = stress_new.deviatoric();
+      const Real norm_dev_stress =
+          std::sqrt(deviatoric_stress.doubleContraction(deviatoric_stress));
+      mooseAssert(norm_dev_stress != 0.0, "Norm of the deviatoric is zero");
```

Comment:
I think the deviatoric stress would always be nonzero if the effective inelastic strain is nonzero, right?  In that case, I think the assert is the right thing to do here -- an extra precaution, but shouldn't really be necessary.

Reply:
Ah yes, but this is precisely why asserts are great! You are stating your assumption for a piece of code to work. The assumption should hold, but often over time as code is refactored, or a sufficiently talented user feeds it bad inputs, you'll one day get lucky and trigger that assertion saving somebody valuable debugging time. 

They really _should_ never be necessary. They are an explicit statement of an assumption that must hold.

Reply:
👍